core/l10n_init.js

```diff
@@ -61,6 +61,8 @@ define('core/l10n_init',
     }
 
     function _transformLocale(locale) {
+        // Use default locale "en-US" if given locale is undefined (BUG 1138558)
+        locale = locale || 'en-US';
```

Comment:
I'm alright with the fix for ```l10n_init.js```, though I don't think the fix in ```l10n.js``` will change anything in production. I'd just nitpicking since I think ```l10n.js``` shouldn't worry about ```window.navigator.l10n``` so any fall backs should be applied in ```l10n_init.js```